packages/wallet-backend-update/src/user/controller.ts

```diff
@@ -0,0 +1,65 @@
+import { NotFound } from '@/errors'
+import type { NextFunction, Request } from 'express'
+import type { Logger } from 'winston'
+import type { UserService } from './service'
+import type { SessionService } from '@/session/service'
+
+interface UserProfile {
+  email: string
+  lastName?: string
+  firstName?: string
+  address?: string
+  country?: string
+  needsWallet: boolean
+  needsIDProof: boolean
+}
+
+interface IUserController {
+  me: (
```

Comment:
The `/me` endpoint can be hit even if the user is not authenticated - it is not guarded by the `isAuth` middleware.
It has the same behaviour like the current `/me` endpoint on our initial backend - retrieving user information if it exists. We call `session.destroy` to delete all the cookies that the user has - in case there are any.

On the other hand, I think we can simply improve this by returning early if `req.session.id` or `req.session.user` do not exist. By doing this we will not make queries to our database if the session is invalid.

Reply:
Wouldn't it be better to have it guarded by IsAuth, in which case it would return Unauthorised if the req.session.id or req.session.user do not exist?
 
 I don't see a usecase in which you hit the me endpoint as an unauthorised user?